arch/arm/src/nrf52/nrf52_gpio.c

```diff
@@ -204,7 +236,8 @@ int nrf52_gpio_config(nrf52_pinset_t cfgset)
       switch (cfgset & GPIO_FUNC_MASK)
         {
         case GPIO_INPUT:   /* GPIO input pin */
-          break;           /* Already configured */
+          nrf52_gpio_sense(cfgset, port, pin);
```

Comment:
This triggers GPIOTE events even if we haven't enabled pin events yet.
As I understand, SENSE is only used for GPIOTE, then it should be configured when we enable the given pin event.

Reply:
I'm not sure what you mean by GPIOTE vs pin events.
SENSE is used for waking up the MCU from IDLE sleep (without an extra interrupt) but most importantly from system OFF.
SENSE generates an internal DETECT signal (depending on DETECTMODE register), which in turns generates the PORT event. What controls waking up is the DETECT signal and is independent of handling (or not) the PORT event via interrupt.

The thing about DETECT is that if you enable SENSE_HIGH when the pin is HIGH it will immediately interrupt (as expected). An option to not generate the interrupt is to make the interrupt enable be dependant on having the callback set (as an indication that the user really wants the interrupt to be generated).

Reply:
OK, you're right. I didn't notice that DETECT has more functionality than just a PORT event.